core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4548,29 +4548,42 @@ public Object getTarget() {
         try {
             checkPermission(READ);
         } catch (AccessDeniedException e) {
-            String rest = Stapler.getCurrentRequest().getRestOfPath();
-            for (String name : ALWAYS_READABLE_PATHS) {
-                if (rest.startsWith(name)) {
-                    return this;
-                }
-            }
-            for (String name : getUnprotectedRootActions()) {
-                if (rest.startsWith("/" + name + "/") || rest.equals("/" + name)) {
-                    return this;
-                }
-            }
-
-            // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-            if (rest.matches("/computer/[^/]+/slave-agent[.]jnlp")
-                && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
+            if (isPathUnprotected(Stapler.getCurrentRequest().getRestOfPath())) {
                 return this;
             }
 
-
             throw e;
         }
         return this;
     }
+    
+    /**
+     * Test a path to see if it does not require authentication
```

Comment:
Any recommendations on a better name? I could call it "isPathAccessibleAnonymously", "isAnonymousAccessAllowed", "isPathAccessibleWithoutAuthentication", or invert the return value and call it "doesPathRequireAuthentication".
The only issue I see (with my current knowledge) with the first two is that they are also misleading, since we are not checking if anonymous access is allowed - they are instead checks for paths where access is allowed to any user, and the check normally only occurs if the user does not have permissions to the path they are trying to access, and the secondary use I am seeing it for is for checking first to see if the user can access those paths, before any authentication attempts take place.

Should the comment instead say "Test a path to see if it is accessible regardless of authentication status"?